src/icesat2_tracks/analysis_db/B05_define_angle.py

```diff
@@ -374,143 +204,372 @@ def linear(self, radial_axis="period", ax=None, cbar_flag=True):
+
+        ax1 = F.fig.add_subplot(gs[0:3, :])
+        ax1.tick_params(labelbottom=False)
+
+        weighted_spec = (Gk.gFT_PSD_data * Gk.N_per_stancil).sum(
+            "beam"
+        ) / Gk.N_per_stancil.sum("beam")
+        x_spec = weighted_spec.x / 1e3
+        k = weighted_spec.k
+
+        xlims = x_spec[0], x_spec[-1]
+        clev_spec = np.linspace(-80, (10 * np.log(weighted_spec)).max() * 0.9, 21)
+
+        plt.pcolor(
+            x_spec,
+            k,
+            10 * np.log(weighted_spec),
+            vmin=clev_spec[0],
+            vmax=clev_spec[-1],
+            cmap=cmap_spec,
+        )
 
-    dir_data = (i_dir * i_dir.N_data).sum(["beam_group", "x"]) / i_dir.N_data.sum(
-        ["beam_group", "x"]
-    )
-    lims = (
-        dir_data.k[(dir_data.sum("angle") != 0)][0].data,
-        dir_data.k[(dir_data.sum("angle") != 0)][-1].data,
-    )
+        _title = f"{track_name}\nPower Spectra (m/m)$^2$ k$^{{-1}}$"
+        plt.title(_title, loc="left")
+
+        cbar = plt.colorbar(
+            fraction=0.018, pad=0.01, orientation="vertical", label="Power"
+        )
+        cbar.outline.set_visible(False)
+        clev_ticks = np.round(clev_spec[::3], 0)
+        cbar.set_ticks(clev_ticks)
+        cbar.set_ticklabels(clev_ticks)
+
+        plt.ylabel("wavenumber $k$")
+
+        ax2 = F.fig.add_subplot(gs[3:5, :])
+        ax2.tick_params(labelleft=True)
+
+        dir_data = Gpdf.interp(x=weighted_spec.x).weighted_angle_PDF_smth.T
+
+        angle = Gpdf.angle
+        plt.pcolor(
+            x_spec,
+            angle,
+            dir_data,
+            vmin=clev_angle[0],
+            vmax=clev_angle[-1],
+            cmap=cmap_angle,
+        )
+
+        cbar = plt.colorbar(
+            fraction=0.01, pad=0.01, orientation="vertical", label="Density"
+        )
+        plt.title("Direction PDF", loc="left")
 
-    N_angle = i_dir.angle.size
-    dir_data2 = dir_data
+        plt.xlabel("x (km)")
+        plt.ylabel("angle")
 
-    plot_data = dir_data2 * i_spec.mean("x")
-    plot_data = plot_data.rolling(angle=5, k=10).median()
+        ax2.set_yticks(xticks_pi)
+        ax2.set_yticklabels(xtick_labels_pi)
 
-    plot_data = plot_data.sel(k=slice(lims[0], lims[-1]))
-    xx = 2 * np.pi / plot_data.k
-
-    if np.nanmax(plot_data.data) != np.nanmin(plot_data.data):
-        ax3 = F.fig.add_subplot(gs, polar=True)
-        FP = plot_polarspectra(
-            xx,
-            plot_data.angle,
-            plot_data,
-            lims=None,
-            verbose=False,
-            data_type="fraction",
+        x_ticks = np.arange(0, xlims[-1].data, 50)
+        x_tick_labels, x_ticks = MT.tick_formatter(
+            x_ticks, expt_flag=False, shift=0, rounder=1, interval=2
         )
-        FP.clevs = np.linspace(
-            np.nanpercentile(plot_data.data, 1), np.round(plot_data.max(), 4), 21
+
+        for ax in [ax1, ax2]:
+            update_axes(ax, x_ticks, x_tick_labels, xlims)
```

Comment:
This `update_axes `function seems to be unnecessary.

```suggestion
          ax.set_xticks(x_ticks)
          ax.set_xticklabels(x_tick_labels)
          ax.set_xlim(xlims))
```

Reply:
In the spirit of making the code more readable and seeing fewer lines of code mid script, I thought this was a reasonable abstraction. @hollandjg What do you think?

Reply:
What about moving the `for` loop inside the function?
My personal opinion is that functions that only makes assignments look alike object's constructors functions.

Reply:
Let's leave the function.